libcxx/include/__math/gamma.h

```diff
@@ -55,6 +55,31 @@ inline _LIBCPP_HIDE_FROM_ABI double tgamma(_A1 __x) _NOEXCEPT {
   return __builtin_tgamma((double)__x);
 }
 
+// __lgamma_r
+
+struct __lgamma_result {
+  double __result;
+  int __sign;
+};
+
+#if _LIBCPP_AVAILABILITY_HAS_THREAD_SAFE_LGAMMA
+_LIBCPP_EXPORTED_FROM_ABI __lgamma_result __lgamma_thread_safe_impl(double) _NOEXCEPT;
+
+inline _LIBCPP_HIDE_FROM_ABI __lgamma_result __lgamma_thread_safe(double __d) _NOEXCEPT {
+  return __math::__lgamma_thread_safe_impl(__d);
+}
```

Comment:
What's the reason for having another wrapper here?

Reply:
I wanted to avoid having a function that is sometimes defined inline and sometimes defined in the dylib, since I think it's more difficult to reason about ODR and ABI-wise.